### PR TITLE
Routes in App

### PR DESCRIPTION
**Pull request by maxxbay**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+import SearchForm from 'components/SearchForm/SearchForm'; // Add this import
 
 const App = () => {
   return (
     <>
-      <main className="container">
-        <MovieListPage />
-      </main>
+      <Router>
+        <main className="container">
```
Comment (lenaricabal) [inline src/App.jsx:12]: it's better to move this tag to the page, because for now, your header live in your main

File: src/App.jsx [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+          <Routes>
+            <Route path="/" element={<MovieListPage />}>
+              <Route index element={<SearchForm />} />{' '}
```
Comment (lenaricabal) [inline src/App.jsx:15]: try to avoid strange things like `{' '}`

File: src/Pages/MovieList/MovieListPage.js [hunk 3]
```diff
@@ -55,41 +60,70 @@ const MovieListPage = () => {
...
+      });
+    },
+    [setSearchParams]
+  );
 
   return (
     <>
       {selectedMovie ? (
-        <MovieDetails movie={selectedMovie} onBackClick={handleBackClick} />
+        <MovieDetails movie={selectedMovie} onClose={handleBackClick} />
```
Comment (lenaricabal) [inline src/Pages/MovieList/MovieListPage.js:110]: You need to use <Outlet /> for rendering different components for different paths. For now, when you close movieDetail it is not redirecting to movieListPage it only close movie detail ( url still contains movie id).

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -17,6 +17,15 @@ const Header = ({
...
 
+  const handleSearch = useCallback(
+    searchQuery => {
+      if (typeof onSearch === 'function') {
+        onSearch(searchQuery);
+      }
+    },
+    [onSearch]
+  );
+
```
Comment (lenaricabal) [inline src/components/Header/Header.js:28]: Need to remember that useCallback and useMemo can save time only in rendering phase. But it doesn't come free - it needs more time and memory on initialization. So you need to decide if the profit is bigger than expenses.

File: src/components/MovieDetails/MovieDetails.js [hunk 5]
```diff
@@ -2,11 +2,16 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import './MovieDetails.scss';
 
-const MovieDetails = ({ movie }) => {
+const MovieDetails = ({ movie, onClose }) => {
   if (!movie) {
     return <div className="movie-details">Movie not found</div>;
   }
```
Comment (lenaricabal) [inline src/components/MovieDetails/MovieDetails.js:8]: It should not be here. Do this in parent component - it has all info you need. Don't even start to render component if there is no content.

File: src/components/MovieDetails/MovieDetails.js [hunk 6]
```diff
@@ -17,7 +22,7 @@ const MovieDetails = ({ movie }) => {
   } = movie;
 
   return (
-    <div className="movie-details">
+    <div className="movie-details" onClick={handleMovieDetailsClick}>
```
Comment (lenaricabal) [inline src/components/MovieDetails/MovieDetails.js:25]: Clickable <div> is a bad practice
You don't need to change this component props - it should not have close callback. Find another, more explicit solution for closing details.

File: src/components/MovieDetails/MovieDetails.js [hunk 7]
```diff
@@ -17,7 +22,7 @@ const MovieDetails = ({ movie }) => {
...
       <img
         className="movie-details__image"
         src={`https://image.tmdb.org/t/p/w185${imageUrl}`}
```
Comment (lenaricabal) [inline src/components/MovieDetails/MovieDetails.js:28]: src="https://image.tmdb.org/t/p/w185https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"

this is result of this solution - duplication

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  const { movieId } = useParams();
+  const navigate = useNavigate();
+  const [movie, setMovie] = useState(null);
+
+  const url = `http://localhost:4000/movies/${movieId}`;
+  const params = useMemo(() => {
+    return {};
+  }, []);
+
+  const { data: fetchedMovie } = useFetch(url, params);
```
Comment (lenaricabal) [inline src/components/MovieDetailsWrapper/MovieDetailsWrapper.js:17]: you have loading and error. Use it in this component

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+const MovieDetailsWrapper = () => {
+  const { movieId } = useParams();
+  const navigate = useNavigate();
+  const [movie, setMovie] = useState(null);
+
+  const url = `http://localhost:4000/movies/${movieId}`;
+  const params = useMemo(() => {
+    return {};
+  }, []);
```
Comment (lenaricabal) [inline src/components/MovieDetailsWrapper/MovieDetailsWrapper.js:15]: What is this??? No sense, just wasting memory) Just make params in useFetch optional with default value `{}`

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import useFetch from '../../Hooks/useFetch';
+import MovieDetails from '../MovieDetails/MovieDetails';
+import './MovieDetailsWrapper.scss';
+
+const MovieDetailsWrapper = () => {
+  const { movieId } = useParams();
+  const navigate = useNavigate();
+  const [movie, setMovie] = useState(null);
```
Comment (lenaricabal) [inline src/components/MovieDetailsWrapper/MovieDetailsWrapper.js:10]: Do you really need this state here? All you need is already in useFetch

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    return {};
+  }, []);
+
+  const { data: fetchedMovie } = useFetch(url, params);
+
+  useEffect(() => {
+    if (!Array.isArray(fetchedMovie)) {
+      setMovie(fetchedMovie);
+    }
+  }, [fetchedMovie]);
```
Comment (lenaricabal) [inline src/components/MovieDetailsWrapper/MovieDetailsWrapper.js:23]: you don't need this useEfect here. And avoid using` if (not something) `.

File: src/Pages/MovieList/MovieListPage.js [hunk 12]
```diff
@@ -32,7 +37,7 @@ const MovieListPage = () => {
   const url = 'http://localhost:4000/movies';
```
Comment (lenaricabal) [inline src/Pages/MovieList/MovieListPage.js:37]: This url is used in many places. Need to store it as constant and import where it is necessary.

File: src/Pages/MovieList/MovieListPage.js [hunk 13]
```diff
@@ -55,41 +60,70 @@ const MovieListPage = () => {
...
-  const handleSubmit = movie => {
+  const handleSubmit = useCallback(movie => {
     console.log('Submitted movie:', movie);
     setIsDialogOpen(false);
-  };
-  const handleSearchQueryChange = value => {
-    setSearchQuery(value);
-    resetPagination();
-  };
+  }, []);
```
Comment (lenaricabal) [inline src/Pages/MovieList/MovieListPage.js:75]: why you are using useCallback in every function?

File: src/components/MovieTile/MovieTile.js [hunk 14]
```diff
@@ -21,6 +24,7 @@ const MovieTile = ({ movie, onClick }) => {
   };
 
   const handleTileClick = () => {
+    navigate(`/${movie.id}`);
```
Comment (lenaricabal) [inline src/components/MovieTile/MovieTile.js:27]: you need to store url params before navigation to details. 
Please pay attention to the expected behavior in HW

File: package.json [hunk 15]
```diff
@@ -13,7 +13,7 @@
     "react-dom": "^18.2.0",
     "react-portal": "^4.2.2",
     "react-router": "^6.9.0",
-    "react-router-dom": "^6.9.0",
+    "react-router-dom": "^6.10.0",
     "react-scripts": "^5.0.1",
     "use-deep-compare-effect": "^1.8.1",
```
Comment (lenaricabal) [inline package.json:18]: do you still need this?

File: package.json [hunk 16]
```diff
@@ -82,4 +82,4 @@
     "prop-types": "^15.8.1",
     "webpack": "^5.76.3"
   }
-}
\ No newline at end of file
+}
```
Comment (lenaricabal) [inline package.json:85]: Where are cypress scripts? Cant check e2e test. And check unit test - not all are passing

File: src/App.jsx (hunk 2 above)
Comment (maxxbay) [reply src/App.jsx:15]: It's really strange things

File: package.json (hunk 16 above)
Comment (maxxbay) [reply package.json:86]: Done

File: package.json (hunk 15 above)
Comment (maxxbay) [reply package.json]: Done

File: src/Pages/MovieList/MovieListPage.js (hunk 13 above)
Comment (maxxbay) [reply src/Pages/MovieList/MovieListPage.js]: Done

File: src/Pages/MovieList/MovieListPage.js (hunk 12 above)
Comment (maxxbay) [reply src/Pages/MovieList/MovieListPage.js]: Done

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js (hunk 11 above)
Comment (maxxbay) [reply src/components/MovieDetailsWrapper/MovieDetailsWrapper.js]: Done

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js (hunk 10 above)
Comment (maxxbay) [reply src/components/MovieDetailsWrapper/MovieDetailsWrapper.js]: Done

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js (hunk 9 above)
Comment (maxxbay) [reply src/components/MovieDetailsWrapper/MovieDetailsWrapper.js]: Done

File: src/components/MovieDetailsWrapper/MovieDetailsWrapper.js (hunk 8 above)
Comment (maxxbay) [reply src/components/MovieDetailsWrapper/MovieDetailsWrapper.js]: Done

File: src/components/MovieDetails/MovieDetails.js (hunk 6 above)
Comment (maxxbay) [reply src/components/MovieDetails/MovieDetails.js]: Done

File: src/components/Header/Header.js (hunk 4 above)
Comment (maxxbay) [reply src/components/Header/Header.js:25]: Done

File: src/Pages/MovieList/MovieListPage.js (hunk 3 above)
Comment (maxxbay) [reply src/Pages/MovieList/MovieListPage.js]: Done

File: src/App.jsx (hunk 1 above)
Comment (maxxbay) [reply src/App.jsx]: Done

**Pull request closed without merging**